source/geometry_model/spherical_shell.cc

```diff
@@ -68,14 +71,38 @@ namespace aspect
       }
 
 
+
+      template <int dim>
+      Point<dim>
+      SphericalManifoldWithTopography<dim>::
+      push_forward_from_sphere(const Point<dim> &p) const
+      {
+        return p;
+      }
+
+
+
+      template <int dim>
+      Point<dim>
+      SphericalManifoldWithTopography<dim>::
+      pull_back_to_sphere(const Point<dim> &p) const
+      {
+        return p;
+      }
```

Comment:
This is where the magic will happen. We just need to stretch in radial direction depending on a point's topography at that lat-long coordinates.